### DISCUSSION TITLE
Error when compiling with DotNet 4.8 and DotNet 8 in same project

### DISCUSSION THREAD

**Discussion by kikmon** [xunit/xunit#3072] [Question] [answered]
Hi

Sorry if I missed the obvious, but I'm try to compile a very simple test project that supports several dotnet framework, including 4.8, through the use of TargetFrameworks tag in the csproj as in 
<TargetFrameworks>net6.0;net8.0;net9.0;net48</TargetFrameworks>

When compiling, net48 is complaining about a 'Feature 'global using directive' is not available in C# 7.3. Please use language version 10.0 or greater.' error, even if 'global includes' setting doesn't appear anywhere in the project. (error coming from a generated .GlobalUsings.g.cs file)

Is this a supported scenario for XUnit ?

Thanks

**Comment by bradwilson** [comment, marked as answer]
Yes, this is supported.

The error you're receiving has nothing to do with xUnit.net, but rather the fact that you're using .NET Framework (net48) and you have enabled global using directives. Since .NET Framework compiles with C# 7.3 by default, you have two ways to resolve this:

1. Disable global usings by removing the property `<ImplicitUsings>enable</ImplicitUsings>` from your .csproj file
2. Update the C# language by adding the property `<LangVersion>10</LangVersion>` to your .csproj file

**Reply by kikmon** [reply]
Thanks for the followup.
'global usings' were already removed from the csproj, so something else is going on.
That being said, I didn't realize I could just update the C# version for 4.8, and it did solve the issue, thanks.
Now my last issue is intellisense not detecting properly me switching between 4.8 and 8, but that's not Xunit related :)

Thanks again
